src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -145,3 +147,56 @@ export async function _readFiles(
 
   return map;
 }
+
+// Checks the provided URL will parse and has no trailing slashes, search or hash.
+// Returns the validated URL without trailing slashes or throws.
+export function _validateUrl(remoteUrl: string): string {
+  try {
+    const url = new global.URL(remoteUrl);
+    if (url.search) {
+      throw `URL should not have a search component: ${url.search}`;
+    }
+    if (url.hash) {
+      throw `URL should not have a hash component: ${url.hash}`;
+    }
+    return url.toString().replace(/\/+$/, "");
+  } catch (e) {
+    throw new Error(
+      `Provided initiatives directory URL was invalid: ${remoteUrl}\n${e}`
+    );
+  }
+}
+
+// Takes a fileName, InitiativeFile map and converts it to Initiatives.
+export function _convertToInitiatives(
+  remoteUrl: string,
+  map: NamesToInitiativeFiles
+): $ReadOnlyArray<Initiative> {
+  const initiatives = [];
+  for (const [fileName, initiativeFile] of map.entries()) {
+    const {timestampIso, ...partialInitiativeFile} = initiativeFile;
+    const initiative: Initiative = {
+      ...partialInitiativeFile,
+      id: createId(INITIATIVE_FILE_SUBTYPE, remoteUrl, fileName),
```

Comment:
Hmm, I'm not very comfortable with this -- it buries the details on how we address the initiatives inside the implementation details of a helper function.

Please factor out a `initiativeId(initiativesDirectory, initiativeMapEntry)` style function that returns the id.

Reply:
Good find. I had already factored this function out in #1643, now updated to use it.